lib/pure/unicode.nim

```diff
@@ -1489,6 +1502,130 @@ proc capitalize*(s: string): string {.noSideEffect, procvar,
 
   result = $toUpper(rune) & substr(s, i)
 
+proc normalizeUnicode*(s: string): string {.noSideEffect,
+  rtl, extern: "nucNormalize".} =
+  ## Normalizes the string ``s``.
+  ##
+  ## That means to convert it to lower case and remove any ``'_'``.
+  const
+    underscore = '_'.Rune
+  var
+    i = 0
+    written = 0
+    rune: Rune
+  result = newString(len(s))
+  while i < len(s):
+    fastRuneAt(s, i, rune)
+    if rune.isUpper():
+      rune = rune.toLower()
+    if rune != underscore:
+      rune.fastToUTF8Copy(result, written)
+
+proc strip*(s: string, leading = true, trailing = true,
+  runes: openarray[Rune] = spaceRanges): string {.noSideEffect,
+  rtl, extern: "nucStrip".} =
+  var
+    s_i = 0
+    e_i = len(s) - 1
+  if leading:
+    var
+      i = 0
+      l_i: int
+      rune: Rune
+    while i < len(s):
+      l_i = i
+      fastRuneAt(s, i, rune)
+      s_i = i # Assume to start from next rune
+      if not runes.contains(rune):
+        s_i = l_i # Go back to where the current rune starts
+        break
+  if trailing:
+    var
+      i = e_i
+      l_i: int
+      rune: Rune
+    while i >= 0:
+      l_i = i
+      fastRuneAt(s, l_i, rune)
+      var p_i = i - 1
+      while p_i >= 0:
+        var
+          p_i_end = p_i
+          p_rune: Rune
+        fastRuneAt(s, p_i_end, p_rune)
+        if p_i_end < l_i: break
+        i = p_i
+        rune = p_rune
+        dec(p_i)
+      if not runes.contains(rune):
+        e_i = l_i - 1
+        break
+      dec(i)
+  let newLen = e_i - s_i
+  result = newStringOfCap(newLen)
+  if e_i > s_i:
+    result.add s[s_i .. e_i]
+
+proc repeat*(c: Rune, count: Natural): string {.noSideEffect,
+  rtl, extern: "nucRepeatRune".} =
+  ## Returns a string of `count` Runes `c`.
+  ##
+  ## The returned string will have a rune-length of `count`.
+  let s = $c
+  result = newStringOfCap(count * s.len)
+  for i in 0 ..< count:
+    result.add s
+
+proc align*(s: string, count: Natural, padding: Rune = ' '.Rune): string {.
+  noSideEffect, rtl, extern: "nucAlignString".} =
+  ## Aligns a unicode string `s` with `padding`, so that it has a rune-length 
+  ## of `count`.
+  ##
+  ## `padding` characters (by default spaces) are added before `s` resulting in
+  ## right alignment. If ``s.runelen >= count``, no spaces are added and `s` is
+  ## returned unchanged. If you need to left align a string use the `alignLeft
+  ## proc <#alignLeft>`_. Example:
+  ##
+  ## .. code-block:: nim
+  ##   assert align("abc", 4) == " abc"
```

Comment:
Some examples with unicode would be nice here.

Reply:
added